lib/maxwell/middleware/header.ex

```diff
@@ -20,20 +20,17 @@ defmodule Maxwell.Middleware.Headers do
   alias Maxwell.Conn
 
   def init(headers) do
-    check_headers(headers)
-    Conn.new()
-    |> Conn.put_req_headers(headers)
-    |> Map.get(:req_headers)
+    check_headers!(headers)
+    headers
```

Comment:
Should use put_req_headers to unify the format or headers, or else might break the internal storage format.